modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/manager/DefaultExtensionManager.java

```diff
@@ -220,25 +214,15 @@ else if (matches > 1)
         return extensionRegistry.getConfigurationProvider(configurationProviderName);
     }
 
-    private boolean attemptToCreateImplicitConfiguration(ExtensionModel extensionModel, MuleEvent muleEvent)
+    private void createImplicitConfiguration(ExtensionModel extensionModel, MuleEvent muleEvent)
     {
         synchronized (extensionModel)
         {
             //check that another thread didn't beat us to create the instance
-            if (!extensionRegistry.getConfigurationProviders(extensionModel).isEmpty())
+            if (extensionRegistry.getConfigurationProviders(extensionModel).isEmpty())
```

Comment:
The previous case always returned true, since the created `configurationInstance` was never null, either one was created, or an exception was thrown. Now, instead of `attemp...` returning false and throwing an exception later, we just propagate the original exception (found [here](https://github.com/mulesoft/mule/pull/2653/commits/e0b8f1b993cb40fe1e8edad66c38fca15089bbf0#diff-edecab14dce16d65b4e6ac6d5eb80df8R41)). 

Also, instead of generating just the instance and forcing the StaticConfigProvider, we now delegate to the factory the Provider creation, which [can also create a Dynamic provider](https://github.com/mulesoft/mule/pull/2653/commits/e0b8f1b993cb40fe1e8edad66c38fca15089bbf0#diff-edecab14dce16d65b4e6ac6d5eb80df8R54) if the configuration is dynamic.
Having the dynamic provider created implicitly allowed us to identify dynamic configs if metadata was invoked using an implicit configuration that contained parameters with dynamic default values.